### ISSUE TITLE
MPS incorrect behavior of torch.nn.functional.pad

### ISSUE THREAD

**Issue by edvul** [pytorch/pytorch#80192]
### 🐛 Describe the bug

**`pad()` on mps device adds padding incorrectly:**

```python
import torch
y = torch.ones(2,1, device='mps')
y_pad = torch.nn.functional.pad(y, (1, 1, 1, 1), 'constant', 0)
print(y_pad)
```

yields:

```
tensor([[0., 0., 0.],
        [0., 1., 1.],
        [0., 0., 0.],
        [0., 0., 0.]], device='mps:0')
```

_**Random observations:**_

1. This appears to be incorrect behavior when falling back on CPU, as the first time running `.pad()` yields the following warning:

```
/opt/homebrew/Caskroom/miniforge/base/envs/py310_m1/lib/python3.10/site-packages/torch/_tensor_str.py:103: UserWarning: The operator 'aten::bitwise_and.Tensor_out' is not currently supported on the MPS backend and will fall back to run on the CPU. This may have performance implications. (Triggered internally at /Users/runner/work/pytorch/pytorch/pytorch/aten/src/ATen/mps/MPSFallback.mm:11.)
  nonzero_finite_vals = torch.masked_select(tensor_view, torch.isfinite(tensor_view) & tensor_view.ne(0))
```

However, running directly on cpu yields the expected result:

```python
y = torch.ones(2,1, device='cpu')
y_pad = torch.nn.functional.pad(y, (1, 1, 1, 1), 'constant', 0)
print(y_pad)
```

returns: 

```
tensor([[0., 0., 0.],
        [0., 1., 0.],
        [0., 1., 0.],
        [0., 0., 0.]])
```

2. this problem arises only when `y.size()[0] > 1`  

```python
y = torch.ones(1,2, device='mps')
y_pad = torch.nn.functional.pad(y, (1, 1, 1, 1), 'constant', 0)
print(y_pad)
```

yields:

```
tensor([[0., 0., 0., 0.],
        [0., 1., 1., 0.],
        [0., 0., 0., 0.]], device='mps:0')
```

### Versions

Collecting environment information...
PyTorch version: 1.13.0.dev20220623
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 12.4 (arm64)
GCC version: Could not collect
Clang version: 12.0.5 (clang-1205.0.22.11)
CMake version: Could not collect
Libc version: N/A

Python version: 3.10.2 | packaged by conda-forge | (main, Feb  1 2022, 19:28:34) [Clang 11.1.0 ] (64-bit runtime)
Python platform: macOS-12.4-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.22.3
[pip3] torch==1.13.0.dev20220623
[pip3] torch-tb-profiler==0.4.0
[pip3] torchaudio==0.14.0.dev20220603
[pip3] torchvision==0.14.0.dev20220623
[conda] numpy                     1.22.3          py310hdb36b11_0  
[conda] numpy-base                1.22.3          py310h5e3e9f0_0  
[conda] torch                     1.13.0.dev20220623          pypi_0    pypi
[conda] torch-tb-profiler         0.4.0                    pypi_0    pypi
[conda] torchaudio                0.14.0.dev20220603          pypi_0    pypi
[conda] torchvision               0.14.0.dev20220623          pypi_0    pypi
(py310_m1) 

cc @ezyang @gchanan @zou3519 @albanD @mruberry @jbschlosser @walterddr @kshitij12345 @saketh-are @kulinseth

**Comment by DenisVieriu97** [comment]
> ```python
> y = torch.ones(2,1, device='mps')
> y_pad = torch.nn.functional.pad(y, (1, 1, 1, 1), 'constant', 0)
> ```

Hi @edvul - thank you for reporting this problem!
This issue is addressed in https://github.com/pytorch/pytorch/pull/79939 (currently in PR) - once it lands in master this should get fixed.

**Comment by DenisVieriu97** [comment]
@edvul The fix from https://github.com/pytorch/pytorch/pull/79939 landed and the latest pytorch nightly build already contains it. Please let me know if you still see the issue. Thanks!
If using pip:
```pip3 install --pre torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/nightly/cpu```

**Comment by kulinseth** [comment]
Thanks Denis . Please reopen if you still see the issue.
